src/main/java/org/checkerframework/checker/androidresource/AndroidResourceAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,402 @@
+                    annotatedTypeMirror.replaceAnnotation(ARRAY_RES);
+
+                } else if (expressionContainerAnnotation.equals(ATTR_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(ATTR_RES);
+
+                } else if (expressionContainerAnnotation.equals(BOOL_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(BOOL_RES);
+
+                } else if (expressionContainerAnnotation.equals(COLOR_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(COLOR_RES);
+
+                } else if (expressionContainerAnnotation.equals(DIMEN_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(DIMEN_RES);
+
+                } else if (expressionContainerAnnotation.equals(DRAWABLE_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(DRAWABLE_RES);
+
+                } else if (expressionContainerAnnotation.equals(FRACTION_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(FRACTION_RES);
+
+                } else if (expressionContainerAnnotation.equals(ID_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(ID_RES);
+
+                } else if (expressionContainerAnnotation.equals(INTEGER_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(INTEGER_RES);
+
+                } else if (expressionContainerAnnotation.equals(INTERPOLATOR_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(INTERPOLATOR_RES);
+
+                } else if (expressionContainerAnnotation.equals(LAYOUT_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(LAYOUT_RES);
+
+                } else if (expressionContainerAnnotation.equals(MENU_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(MENU_RES);
+
+                } else if (expressionContainerAnnotation.equals(PLURALS_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(PLURALS_RES);
+
+                } else if (expressionContainerAnnotation.equals(RAW_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(RAW_RES);
+
+                } else if (expressionContainerAnnotation.equals(STRING_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(STRING_RES);
+
+                } else if (expressionContainerAnnotation.equals(STYLEABLE_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(STYLEABLE_RES);
+
+                } else if (expressionContainerAnnotation.equals(XML_CONTAINER)) {
+                    annotatedTypeMirror.replaceAnnotation(XML_RES);
+                }
+
+            }
+
+
+        }
+
+        /**
+         * Assigning @XXXContainer annotations to member select.
+         */
+        private void assignContainerAnnotations(AnnotatedTypeMirror annotatedTypeMirror, String expression, String identifier) {
+            if (expression.equals("R")) {

```

Comment:
It would again make the code a bit easier to read if you just return if the receiver is not `R`.

Reply:
Fixed.

Reply:
Fixed.